File: Chapter 03/Python/section_3_3_3_lsp/after/forward.py
from .player import Player

"""
LSP SOLUTION: True Substitutability.
* ARCHITECTURE NOTE: The Forward is a true substitute for a Player. It 
doesn't provide "surprising" behavior or refuse the contract. When the 
Coach expects a field action, the Forward delivers exactly that.
"""
class Forward(Player):
    def play_field_position(self):
        print("  [Forward] Leading the attack and trying to score.")

File: Chapter 03/JS/section_3_3_3_lsp/after/forward.js
const Player = require('./player');

/**
 * LSP SOLUTION: True Substitutability.
 * * ARCHITECTURE NOTE: The Forward is a true substitute for a Player. It 
 * doesn't provide "surprising" behavior or refuse the contract. When the 
 * Coach expects a field action, the Forward delivers exactly that.
 */
class Forward extends Player {
    playFieldPosition() {
        console.log("  [Forward] Leading the attack and trying to score.");
    }
}

module.exports = Forward;